quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/core/properties/QuarkusConfigMappingProvider.java

```diff
@@ -0,0 +1,303 @@
+ * interfaces annotated with "io.smallrye.config.ConfigMapping" annotation.
+ * 
+ * @author Angelo ZERR
+ * 
+ * @see https://quarkus.io/guides/config-mappings
+ */
+public class QuarkusConfigMappingProvider extends AbstractAnnotationTypeReferencePropertiesProvider {
+
+	private static final Logger LOGGER = Logger.getLogger(QuarkusConfigMappingProvider.class.getName());
+
+	private static final String[] ANNOTATION_NAMES = { CONFIG_MAPPING_ANNOTATION };
+
+	@Override
+	protected String[] getAnnotationNames() {
+		return ANNOTATION_NAMES;
+	}
+
+	@Override
+	protected void processAnnotation(IJavaElement javaElement, IAnnotation annotation, String annotationName,
+			SearchContext context, IProgressMonitor monitor) throws JavaModelException {
+		processConfigMapping(javaElement, annotation, context.getCollector(), monitor);
+	}
+
+	// ------------- Process Quarkus ConfigMapping -------------
+
+	private void processConfigMapping(IJavaElement javaElement, IAnnotation configMappingAnnotation,
+			IPropertiesCollector collector, IProgressMonitor monitor) throws JavaModelException {
+		if (javaElement.getElementType() != IJavaElement.TYPE) {
+			return;
+		}
+		IType configMappingType = (IType) javaElement;
+		if (!configMappingType.isInterface()) {
+			// @ConfigMapping can be used only with interfaces.
+			return;
+		}
+		// Location (JAR, src)
+		IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) javaElement
+				.getAncestor(IJavaElement.PACKAGE_FRAGMENT_ROOT);
+		String location = packageRoot.getPath().toString();
+		// Quarkus Extension name
+		String extensionName = JDTQuarkusUtils.getExtensionName(location);
+
+		String prefix = determinePrefix(configMappingType, configMappingAnnotation);
+		List<IType> allInterfaces = new ArrayList<>(Arrays.asList(findInterfaces(configMappingType, monitor)));
+		allInterfaces.add(0, configMappingType);
+		for (IType configMappingInterface : allInterfaces) {
+			populateConfigObject(configMappingInterface, prefix, extensionName, new HashSet<>(),
+					configMappingAnnotation, collector, monitor);
+		}
+	}
+
+	private static IType[] findInterfaces(IType type, IProgressMonitor progressMonitor) throws JavaModelException {
+		ITypeHierarchy typeHierarchy = type.newSupertypeHierarchy(progressMonitor);
+		return typeHierarchy.getAllSuperInterfaces(type);
+	}
+
+	private void populateConfigObject(IType configMappingType, String prefixStr, String extensionName,
+			Set<IType> typesAlreadyProcessed, IAnnotation configMappingAnnotation, IPropertiesCollector collector,
+			IProgressMonitor monitor) throws JavaModelException {
+		if (typesAlreadyProcessed.contains(configMappingType)) {
+			return;
+		}
+		typesAlreadyProcessed.add(configMappingType);
+		IJavaElement[] elements = configMappingType.getChildren();
+		// Loop for each methods
+		for (IJavaElement child : elements) {
+			if (child.getElementType() == IJavaElement.METHOD) {
+				IMethod method = (IMethod) child;

```

Comment:
You asked if we could get the information about generics.

Supposed we have `List<String>`, then `method.getReturnType()` gives `QArrayList<QString;>;` . It's in there, but I haven't found a utility method to get the generic element (`QString`) but I'm pretty sure it should be possible.

Reply:
Thanksforthe feedback!